sdk/core/Azure.Core.Experimental/src/DynamicJsonOptions.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.Dynamic
+{
+    /// <summary>
+    /// Provides the ability for the user to define custom behavior when accessing JSON through a dynamic layer.
+    /// </summary>
+    public struct DynamicJsonOptions
+    {
+        /// <summary>
+        /// Creates a new instance of DynamicJsonOptions.
+        /// </summary>
+        public DynamicJsonOptions() { }
+
+        /// <summary>
+        /// Specifies how properties on <see cref="DynamicJson"/> will get accessed in the underlying JSON buffer.
+        /// </summary>
+        public DynamicJsonPropertyCasing PropertyCasing { get; set; } = DynamicJsonPropertyCasing.Default;
+    }
+
+    /// <summary>
+    /// Casing options for property access on DynamicJson.
+    /// </summary>
+    public struct DynamicJsonPropertyCasing
```

Comment:
These types will move to separate files if we go with this approach.

Reply:
Can this be simply an enum?

Reply:
I like the names you suggested in your enum, and if we went with something like that we could.  Thanks for the suggestion!